### ISSUE TITLE
Fail to pull the docker image with an error "failed to register layer: Cannot create layer with missing parent"

### ISSUE THREAD

**Issue by v-mengwe** [docker/for-win#13356]
- Windows Version: Windows 10 Enterprise 22H2
  - Docker Desktop Version: **4.18.0(103181)**

**Repro Steps:**
Run command: docker pull mcr.microsoft.com/dotnet/framework/aspnet:4.8-windowsservercore-ltsc2019

**Expect:**
Succeed to pull the docker image.

**Actual:**
Fail to pull the docker image with an error.
![image](https://user-images.githubusercontent.com/62053665/229438632-11e2d478-7852-4af6-b2f1-5dfe234535a7.png)

**Comment by v-mengwe** [comment]
This issue also reproduces on Docker Desktop 4.17.1.

**Comment by dam-pav** [comment]
Still happens, now on Docker Desktop 4.27.2.

Actually, it started happening when I uninstalled 4.27.2 due to some other issues related to PS handling, tried the installation of a bunch of older versions and finally ended up back with 4.27.2 and a hotfix. Apparently, uninstallation alone does neither reset the configuration completely (it does remove your containers), nor keep it in working order.

I found this enlightening: [Docker on Windows Server 2019: failed to register layer: Cannot create layer with missing parent](https://stackoverflow.com/questions/68873134/docker-on-windows-server-2019-failed-to-register-layer-cannot-create-layer-wit)

Long story short, in Docker Desktop, got to Troubleshooting, select Clean / Purge Data, select Windows Containers, confirm Delete.

**Comment by dirkmass** [comment]
I have the same issue here with 4.36 and also with 4.39. Tried to delete the folder doesn't work. Reindtalling Docker also doesn't work. I tried it with different microsoft business central artifacts.
